packaging/linux/snap/package.mak

```diff
@@ -0,0 +1,10 @@
+snap:
+	snapcraft prime
+	cp ../../../icons/shotcut-logo-64.png gui
+	cp ../org.shotcut.Shotcut.desktop gui/shotcut.desktop
+	desktop-file-edit --set-key=Icon --set-value='${SNAP}/meta/gui/shotcut-logo-64.png' gui/shotcut.desktop
+	snapcraft snap
+
+snap-clean:
+	snapcraft clean
+	rm -rf snap
```

Comment:
desktop-file-edit fails with an error:
```
snap/gui/shotcut.desktop: error: value "${SNAP}/meta/gui/shotcut-logo-64.png" for key "Icon" in group "Desktop Entry" looks like a relative path, instead of being an absolute path to an icon or an icon name
```

Reply:
This approach with snapcraft prime and snap works for the icon (if you target the directory prime/meta/gui), but snapcraft is deleting the desktop file from both the prime dir and the installed snap! Otherwise, the ideal place for these commands is in an [override-stage](https://docs.snapcraft.io/build-snaps/scriptlets) key in the part.

However, I am struggling to get that to work. What does work is moving package.mak into the parent directory with this:
```
snap:
	mkdir -p snap/meta/gui
	cp ../../icons/shotcut-logo-64.png snap/gui
	cp org.shotcut.Shotcut.desktop snap/gui/shotcut.desktop
	sed -i 's|Icon=.*|Icon=${SNAP}/meta/gui/shotcut-logo-64.png|' snap/gui/shotcut.desktop
	snapcraft
```
Nice thing about that is I can add a target for the appimage as well.